skglm/experimental/sqrt_lasso.py

```diff
@@ -0,0 +1,117 @@
+import warnings
+import numpy as np
+from numpy.linalg import norm
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model._base import LinearModel, RegressorMixin
+
+from skglm.penalties import L1
+from skglm.utils import compiled_clone
+from skglm.datafits.base import BaseDatafit
+from skglm.solvers.prox_newton import prox_newton
+
+
+class SqrtQuadratic(BaseDatafit):
+    """Square root quadratic datafit.
+
+    The datafit reads::
+        ||y - Xw||_2 / sqrt(n_samples) 
+    """
+
+    def __init__(self):
+        pass
+
+    def get_spec(self):
+        spec = ()
+        return spec
+
+    def params_to_dict(self):
+        return dict()
+
+    def value(self, y, w, Xw):
+        return np.linalg.norm(y - Xw) / np.sqrt(len(y))
+
+    def raw_grad(self, y, Xw):
+        """Compute gradient of datafit w.r.t ``Xw``.
+
+        Raises:
+        -------
+            Exception
+                if value of residuals is too small (less than ``1e-10``).
+        """
+        minus_residual = Xw - y
+        norm_residuals = norm(minus_residual)
+
+        if norm_residuals < 1e-10:
```

Comment:
we should aim at making the threshold homogeneous in y (avoid values that are not relative to the data, when possible)

alo, in the snippet I pushed, it takes 700 PN iterations to reach this warning: it means the threshold is too low.

can you try using `1e-3 * norm(y)` ? Or even 1e-2 ?

Reply:
that way, we need to compute norm y every time we call ``raw_grad``, WDYT?

Reply:
I think it's still better than having a solver running for 700 iterations before realizing it is stuck
The cost is of the same order of computing the gradient, it's OK: go for the norm recomputation